File: src/app/api/requests/create-effect-request.ts
import {IStatModifier} from '../shared';
import {DurationType} from '../shared/enums';

export interface CreateEffectRequest {
    name: string;
    description: string;
    dice: number | null;
    durationType: DurationType;
    duration: string;
    combatCount: number | null;
    lapCount: number | null;
    timeDuration: number | null;
    modifiers: IStatModifier[];
}

File: src/Naheulbook.Requests/Requests/CreateEffectRequest.cs
using System.Collections.Generic;
using Naheulbook.Shared.TransientModels;

namespace Naheulbook.Requests.Requests;

[PublicAPI]
public class CreateEffectRequest
{
    public required string Name { get; set; }
    public string? Description { get; set; }
    public required string DurationType { get; set; }
    public int? CombatCount { get; set; }
    public string? Duration { get; set; }
    public short? Dice { get; set; }
    public int? LapCount { get; set; }
    public int? TimeDuration { get; set; }
    public required List<StatModifierRequest> Modifiers { get; set; }
}